tests/suppressionlist_test.php

```diff
@@ -0,0 +1,174 @@
+     * to return predefined test data. It then executes the task and
+     * verifies that the suppression list in the database is updated correctly.
+     *
+     * @return void
+     * @covers \tool_emailutils\task\update_suppression_list::execute
+     * @covers \tool_emailutils\task\update_suppression_list::fetch_aws_ses_suppression_list
+     * @throws dml_exception
+     */
+    public function test_update_suppression_list_task(): void {
+        global $DB;
+        $this->resetAfterTest(true);
+
+        // Set up mock AWS credentials (these won't be used, but let's set them anyway).
+        set_config('aws_region', 'eu-west-2', 'tool_emailutils');
+        set_config('aws_key', 'testkey', 'tool_emailutils');
+        set_config('aws_secret', 'testsecret', 'tool_emailutils');
+
+        $mocktask = new class extends \tool_emailutils\task\update_suppression_list {
+            /**
+             * Override the execute method to use our mocked data.
+             *
+             * @return void
+             */
+            public function execute(): void {
+                global $DB;
+                $suppressionlist = $this->fetch_aws_ses_suppression_list();
+                $DB->delete_records('tool_emailutils_suppression');
+                foreach ($suppressionlist as $item) {
+                    $record = new \stdClass();
+                    $record->email = $item['email'];
+                    $record->reason = $item['reason'];
+                    $record->created_at = $item['created_at'];
+                    $record->timecreated = time();
+                    $record->timemodified = time();
+                    $DB->insert_record('tool_emailutils_suppression', $record);
+                }
+            }
+
+            /**
+             * Override the fetch method to return mock data.
+             *
+             * @return array Mock suppression list data.
+             */
+            protected function fetch_aws_ses_suppression_list(): array {
+                return [
+                    [
+                        'email' => 'suppressed@example.com',
+                        'reason' => 'BOUNCE',
+                        'created_at' => '2024-03-03 12:00:00',
+                    ],
+                ];
+            }
+        };
+
+        // Execute the mock task.
+        $mocktask->execute();
+
+        // Check if the suppression list was updated in the database.
+        $record = $DB->get_record('tool_emailutils_suppression', ['email' => 'suppressed@example.com']);
+        $this->assertNotFalse($record);
+        $this->assertEquals('BOUNCE', $record->reason);
+        $this->assertEquals('2024-03-03 12:00:00', $record->created_at);
+    }
+
+    /**
+     * Test AWS credentials configuration.
+     *
+     * This test verifies that AWS credentials can be correctly set and retrieved
+     * from the Moodle configuration.
+     *
+     */
+    public function test_aws_credentials_config() {
```

Comment:
Again, this is just testing the core get/set_config functions, there is no value to the plugin testing them here.